microsoft-edge/webview2/gettingstarted/winui.md

```diff
@@ -0,0 +1,306 @@
+
+*   `SourceChanged`  
+*   `ContentLoading`  
+*   `HistoryChanged`  
+    
+When there is an HTTP redirect, there are multiple `NavigationStarting` events.  
+
+To demonstrate how to use the events, start by registering a handler for `NavigationStarting` event that cancels any requests that do not use HTTPS.  
+
+In `MainWindow.xaml.cs`, modify the constructor as shown below and add the `EnsureHttps` function.  
+
+```csharp
+public MainWindow()
+{
+    InitializeComponent();
+    MyWebView.NavigationStarting += EnsureHttps;
+}
+
+private void EnsureHttps(WebView2 sender, WebView2NavigationStartingEventArgs args)
+{
+    String uri = args.Uri;
+    if (!uri.StartsWith("https://"))
+    {
+        args.Cancel = true;
+    }
+    else
+    {
+        addressBar.Text = uri;
+    }
+}
+```  
+
+In the constructor, EnsureHttps is registered as the event handler on the `NavigationStarting` event on the WebView2 control.  
+
+Press `F5` to build and run your project.  Confirm that when navigating to an HTTP site, the WebView remains unchanged.  However, the WebView navigates to HTTPS sites.  
+
+## Step 5 - Scripting  
+
+You may use host applications to inject JavaScript code into WebView2 controls at runtime.  The injected JavaScript applies to all new top level documents and any child frames until the JavaScript is removed.  The injected JavaScript is run after creation of the global object, and before any other script included in the HTML document is run.  
+
+You may use scripting to alert the user when navigating to a non-HTTPS site.  Modify the `EnsureHttps` function, so that it injects script into the web content using the [ExecuteScriptAsync][Webviews2ReferenceWpf09515MicrosoftWebExecutescriptasync] method.  
+
+```csharp
+private void EnsureHttps(WebView2 sender, WebView2NavigationStartingEventArgs args)
+{
+    String uri = args.Uri;
+    if (!uri.StartsWith("https://"))
+    {
+        MyWebView.ExecuteScriptAsync($"alert('{uri} is not safe, try an https link')");
+        args.Cancel = true;
+    }
+    else
+    {
+        addressBar.Text = uri;
+    }
+}
+```  
+
+Press `F5` to build and run your project.  Confirm that your application displays an alert when you navigate to a site that does not use HTTPS.  
+
+:::image type="complex" source="./media/winui-gettingstarted-script.png" alt-text="HTTP alert" lightbox="./media/winui-gettingstarted-script.png":::
+   HTTP alert  
+:::image-end:::  
+
+> [!NOTE]
+> Congratulations, you built your first WebView2 app.  
+
+## Next Steps  
+
+Our team is currently working on building out the WebView2 APIs to provide full WebView2 functionality.  For more information of the current state of the WebView2 API, see [WebView2 spec][GithubMicrosoftUiXamlSpecsWebview2].  
+
+> [!NOTE]
+> The API may ship before the WinRT CoreWebView2 object, meaning functionality of the Webview2 would be limited to the APIs included in the above document for some interim period of time.  
```

Comment:
Can you clarify the sentence above please? 
"The API may ship..." - which API does this refer to?